### PR TITLE
Rulers

### PR DESCRIPTION
**Pull request by nerik**
Adds the possibility to draw rulers on the map to check distances.
Needs https://github.com/GlobalFishingWatch/map-components/pull/65

Screenshot 2019-07-02 at 13 45 32

TODO:
- [x] check interaction with @satellitestudiodesign 
~- [ ] workspace save/read~
- [x] toolbar UI

### CODE REVIEW

File: src/app/rulers/components/Rulers.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import Icon from 'app/components/Shared/Icon'
+
+import RulersStyles from './rulers.module.scss'
+
+class Rulers extends Component {
+  state = {
```
Comment (j8seangel) [inline src/app/rulers/components/Rulers.jsx]: 🙄 [the copy and paste components caveats]

File: src/app/rulers/rulersActions.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export const EDIT_RULER = 'EDIT_RULER'
+export const MOVE_CURRENT_RULER = 'MOVE_CURRENT_RULER'
+export const TOGGLE = 'TOGGLE'
```
Comment (j8seangel) [inline src/app/rulers/rulersActions.js]: Isn't this action too generic ? also we could have namespacing issues, so I'd prefer having something like `RULER_TOGGLE`

Comment (j8seangel) [reply src/app/rulers/rulersActions.js]: same for `RESET`

File: src/app/rulers/rulersSelectors.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  finalFeature.properties.isNew = isNew
+  return finalFeature
+}
+
+export const getRulersLayer = createSelector(
+  [getRulers, getRulersVisibility],
+  (rulers, rulersVisibility) => {
+    const geojson = {
+      type: 'FeatureCollection',
+      features: rulers.map(makeRulerGeometry),
```
Comment (j8seangel) [inline src/app/rulers/rulersSelectors.js]: Thanks for moving this logic to the selector, IMO it keeps it simpler and makes it easier to maintain

File: src/app/rulers/rulersActions.js (hunk 2 above)
Comment (nerik) [reply src/app/rulers/rulersActions.js]: 👍

**Pull request merged**